### ISSUE TITLE
Test fails with Java 11

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#1070]
```
Failures: 
  JdtTypeProviderTest>AbstractTypeProviderTest.testFindTypeByName_javaLangCharSequence_02:322->AbstractTypeProviderTest.assertMembers:229->Assert.assertEquals:631->Assert.assertEquals:645->Assert.failNotEquals:834->Assert.fail:88 expected:<6> but was:<7>
  SourceBasedJdtTypeProviderTest>AbstractTypeProviderTest.testFindTypeByName_javaLangCharSequence_02:322->AbstractTypeProviderTest.assertMembers:229->Assert.assertEquals:631->Assert.assertEquals:645->Assert.failNotEquals:834->Assert.fail:88 expected:<6> but was:<7>
  ContentAssistTest.testSubtypeArrayList_01:163->Assert.assertEquals:144->Assert.assertEquals:115 expected:<[java.util.ArrayList
java.util.Arrays.ArrayList]> but was:<[]>
  ContentAssistTest.testSubtypeArrayList_02:168->Assert.assertEquals:144->Assert.assertEquals:115 expected:<[ArrayList
Arrays.ArrayList]> but was:<[]>
  ContentAssistTest.testSubtypeBlockingQueue_01:180->Assert.assertEquals:144->Assert.assertEquals:115 expected:<[concurrent.BlockingQueue]> but was:<[]>
  ContentAssistTest.testSubtypeBlockingQueue_02:184->Assert.assertEquals:144->Assert.assertEquals:115 expected:<[util.concurrent.BlockingQueue]> but was:<[]>
  ContentAssistTest.testSubtypeBlockingQueue_03:188->Assert.assertEquals:144->Assert.assertEquals:115 expected:<[util.concurrent.BlockingQueue]> but was:<[]>
  ContentAssistTest.testSubtypeHashSet_01:172->Assert.assertEquals:144->Assert.assertEquals:115 expected:<[java.util.HashSet]> but was:<[]>
  ContentAssistTest.testSubtypeHashSet_02:176->Assert.assertEquals:144->Assert.assertEquals:115 expected:<[HashSet]> but was:<[]>

```
looks like subtype proposals is broken and does not find BlockingQueue
(it does not seem to find anything from JDK)

```
Results:

Failures: 
  JdtTypeProviderTest>AbstractTypeProviderTest.testFindTypeByName_javaLangCharSequence_02:322->AbstractTypeProviderTest.assertMembers:229->Assert.assertEquals:631->Assert.assertEquals:645->Assert.failNotEquals:834->Assert.fail:88 expected:<6> but was:<7>
  SourceBasedJdtTypeProviderTest>AbstractTypeProviderTest.testFindTypeByName_javaLangCharSequence_02:322->AbstractTypeProviderTest.assertMembers:229->Assert.assertEquals:631->Assert.assertEquals:645->Assert.failNotEquals:834->Assert.fail:88 expected:<6> but was:<7>
  ContentAssistTest.testSubtypeHashSet_01:172->Assert.assertEquals:144->Assert.assertEquals:115 expected:<[java.util.HashSet]> but was:<[]>
  ContentAssistTest.testSubtypeHashSet_02:176->Assert.assertEquals:144->Assert.assertEquals:115 expected:<[HashSet]> but was:<[]>

```
```
  Storage2UriMapperJavaImplTest.testOnClasspathChange:114->lambda$3:110->Assert.assertEquals:645->Assert.failNotEquals:834->Assert.fail:88 /home/dietrich/Downloads/jdk-11/lib/jrt-fs.jar expected:<1> but was:<49>
  Storage2UriMapperJavaImplTest.testOnCloseOpenRemoveProject:185->assertBothProjects:263->lambda$13:257->Assert.assertEquals:645->Assert.failNotEquals:834->Assert.fail:88 /home/dietrich/Downloads/jdk-11/lib/jrt-fs.jar expected:<2> but was:<99>
  Storage2UriMapperJavaImplTest.testOnRemoveTwoProjects:211->assertBothProjects:263->lambda$13:257->Assert.assertEquals:645->Assert.failNotEquals:834->Assert.fail:88 /home/dietrich/Downloads/jdk-11/lib/jrt-fs.jar expected:<2> but was:<99>

```
```
Content*Test*.*test*
```
TODO other projects

**Comment by cdietrich** [comment]
@iloveeclipse @stephan-herrmann do you have any hints why (basic)searchengine does not find types from jdk on java 11?

**Comment by iloveeclipse** [comment]
@cdietrich : which target platform is it? On 4.11 we had a bug that the type search was broken, this should be fixed in 4.12 M1.

**Comment by cdietrich** [comment]
/data/.p2/pool/plugins/org.eclipse.jdt.core_3.18.0.v20190430-1056.jar
am not sure if this already has the fix

**Comment by iloveeclipse** [comment]
That one should be OK. I would say if you have a test that only based on JDT/SDK code that fails, create a bug and attach the test to it. I see it has something with "ContentAssist", so may be it is broken somewhere else => we need a reproducer.

**Comment by cdietrich** [comment]
hmmm am not sure if i find the time to create one.
do you know when we need java 11 in customer project? 
that would be a blocker there

**Comment by iloveeclipse** [comment]
We are already working on transition (using Xtext 2.16 as base, probably 2.17 if we will be able to update to it). The plan is to move as soon as we can (but I assume that this will happen in the second half of 2019).

**Comment by cdietrich** [comment]
created https://bugs.eclipse.org/bugs/show_bug.cgi?id=547051

**Comment by cdietrich** [comment]
andrey found and fixed the problem in jdt.
=> we can continue investigating once its available in i-builds

**Comment by cdietrich** [comment]
The fix @iloveeclipse did for https://bugs.eclipse.org/bugs/show_bug.cgi?id=547051
fixes parts of the test. but there are some that still are open.
added another testcase to the bugzilla

**Comment by cdietrich** [comment]
with the latest jdt i-build the content assist tests are fine now.
so the only thing left is the `Storage2UriMapperJavaImplTest`
@szarnekow maybe you have idea how to deal with modular jdk/jre there

**Comment by cdietrich** [comment]
fixed in 2.18/2.19
